django/lego/dns.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""DNS utiilities."""
+
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+from git import Git, Repo
+
+from .settings import DNS_REPOSITORY_URL, SSH_IDENTITY_FILE
+
+REPOSITORY_BASE_URL = "@".join(DNS_REPOSITORY_URL.split("@")[:-1])
+REPOSITORY_BRANCH = DNS_REPOSITORY_URL.split("@")[-1]
+RECORD_CONTENT = "; GENERATED BY httpreq - DO NOT EDIT!"
+SSH_EXECUTABLE = f"ssh -i {SSH_IDENTITY_FILE}"
+
+
+def write_dns_record(dns_record: str) -> None:
```

Comment:
Shouldn't we manage exception that could be raised by this method and return custom errors ?

Reply:
Indeed